r1cs/gadgets/std/src/fields/fp12.rs

```diff
@@ -703,6 +850,10 @@ where
     }
 }
 
+/* enforces that all two components of two Fp12Gadgets are not equal.
```

Comment:
Since that each implementation from Fp2Gadget above uses the underlying FpGadget implementation, there's no need to comment the same thing again. I would remove it here and in all the other files (except fp.rs)